### ISSUE TITLE
"Invalid Host \"2018-03-01 13\"."

### ISSUE THREAD

**Issue by kate-kate** [api-platform/core#2497]
If I send POST request like "/api/jobs" with data:
```json
{
    "timeStart": "2018-03-01 13:01"
}
```
I will receive error with text "Invalid Host \"2018-03-01 13\"."
In entity ```Job``` property ```timeStart``` is type of ```\DateTime```
I made some crazy debug and found, that problem is raising in few steps:
1. AbstractItemNormalizer in setAttributeValue() do this
```php
       if (null !== $className = $type->getClassName()) {
            $this->setValue(
                $object,
                $attribute,
                $this->denormalizeRelation($attribute, $propertyMetadata, $className, $value, $format, $this->createChildContext($context, $attribute))
            );

            return;
        }
```
as timeStart is type of \DateTime, it tries to denormalize relation
2. denormalizeRelation do this:
```php
       if (\is_string($value)) {
            try {
                return $this->iriConverter->getItemFromIri($value, $context + ['fetch_data' => true]);
            } catch (ItemNotFoundException $e) {
                throw new InvalidArgumentException($e->getMessage(), $e->getCode(), $e);
            } catch (InvalidArgumentException $e) {
                // Give a chance to other normalizers (e.g.: DateTimeNormalizer)
            }
        }
```
and first of all tries to get item from iri "2018-03-01 13:01" (he thinks it is an iri)
3. IriConverter tries to create route and then I came into my error with invalid host because it recognize "2018-03-01 13" as host and value after semicolon as port

I can avoid getting this error by sending time with seconds always. But when you get this error it is quite hard to understand, what is going on, so maybe you will decide to avoid it somehow

**Comment by ragboyjr** [comment]
@kate-kate interesting issue, what version of API Platform are you using?

**Comment by kate-kate** [comment]
@ragboyjr 2.3.0

**Comment by ragboyjr** [comment]
Thanks @kate-kate, are you intentionally closing/re-opening this ticket or is that just an accident?

This definitely looks like a bug, will have to do some testing on latest API platform to see if it still persists, I'd imagine some of the work done on the input/output classes ticket could affect this for the 2.4.0 release.

**Comment by norkunas** [comment]
#2298

**Comment by kate-kate** [comment]
@ragboyjr It was accidentally)

**Comment by teohhanhui** [comment]
Fixed in #2514
